### PR TITLE
#246: Playlist update only introduce fields to change

### PR DESCRIPTION
**Pull request by oludaara**
…Remove songs endpoints, and update tests

Description
This pull request refactors the playlist update logic to allow partial updates of playlist metadata, introduces dedicated endpoints for adding and removing songs from playlists, and updates the corresponding backend tests to match the new API structure.

Checklist
<input checked="" disabled="" type="checkbox"> I have read and followed the [Contributing Guidelines](vscode-file://vscode-app/c:/Users/BJ/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).
<input checked="" disabled="" type="checkbox"> I have followed the project's [git convention](vscode-file://vscode-app/c:/Users/BJ/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).
<input checked="" disabled="" type="checkbox"> I have tested the changes and verified they work as intended. Automated tests (unit, integration, e2e as applicable) and manual tests (UI interactions, edge cases, error states, etc.).
<input checked="" disabled="" type="checkbox"> I have set the pull request base to the appropriate development-vx.xx branch corresponding to the issue milestone.

Commit type
feat: indicates the addition of a new feature or functionality to the project.
Issue

(Issue #246 )

Proposed Changes
Added PATCH endpoint for updating playlist metadata with only the fields to be changed (name, description, photo).
Added PATCH and DELETE endpoints for adding and removing songs from playlists.
Updated backend tests to cover the new endpoints and partial update logic.
Refactored service and repository layers to support these changes.
Potential Impact
Frontend must be updated to use the new endpoints for partial updates and song management.
Existing clients using the old full-update pattern may need to be adapted.
No breaking changes to unrelated features.

Assigned
@AntonioMrtz

### CODE REVIEW

File: Backend/app/spotify_electron/playlist/playlist_repository.py [hunk 1]
```diff
@@ -388,3 +388,28 @@ async def remove_songs_from_playlist(name: str, song_names: list[str]) -> None:
         raise PlaylistRepositoryError from exception
     else:
         playlist_repository_logger.info(f"Songs removed from playlist {name}: {song_names}")
+
+
+async def update_playlist_metadata(name: str, update_fields: dict) -> None:
```
Comment (AntonioMrtz) [inline Backend/app/spotify_electron/playlist/playlist_repository.py:393]: dict type should have whats inside it. Example: `dict[str,Any]`

File: Backend/app/spotify_electron/playlist/playlist_service.py [hunk 2]
```diff
@@ -228,6 +228,55 @@ async def update_playlist(  # noqa: PLR0917
         playlist_service_logger.info(f"Playlist {name} updated successfully")
 
 
+async def update_playlist_metadata(
```
Comment (AntonioMrtz) [inline Backend/app/spotify_electron/playlist/playlist_service.py:231]: original `update_playlist` should be gone

File: Backend/app/spotify_electron/playlist/playlist_service.py [hunk 3]
```diff
@@ -228,6 +228,55 @@ async def update_playlist(  # noqa: PLR0917
...
+            update_fields["name"] = new_name
+        if description is not None:
+            update_fields["description"] = description
+        if photo is not None:
+            update_fields["photo"] = photo if "http" in photo else ""
+        if not update_fields:
+            return  # Nothing to update
+        await playlist_repository.update_playlist_metadata(name, update_fields)
+        if new_name:
+            await base_user_service.update_playlist_name(name, new_name)
```
Comment (AntonioMrtz) [inline Backend/app/spotify_electron/playlist/playlist_service.py:256]: This function should have all the logic for the playlist update (even name change)

File: Backend/tests/test__playlist.py [hunk 4]
```diff
@@ -558,3 +559,52 @@ def test_remove_song_from_playlist_song_not_found():
 
     res_delete_artist = delete_user(playlist_owner)
     assert res_delete_artist.status_code == HTTP_202_ACCEPTED
+
+
+def test_update_playlist_metadata():
```
Comment (AntonioMrtz) [inline Backend/tests/test__playlist.py:564]: Adapt regular playlist update tests and add new for edge cases introduced by the new method

File: Backend/tests/test__playlist.py [hunk 5]
```diff
@@ -558,3 +559,52 @@ def test_remove_song_from_playlist_song_not_found():
...
+    # Update only description
+    new_description = "newdesc"
+    res_update = update_playlist_metadata(
+        name=name, descripcion=new_description, headers=jwt_headers
+    )
+    assert res_update.status_code == 204
+    res_get = get_playlist(name=name, headers=jwt_headers)
+    assert res_get.json()["description"] == new_description
+
+    # Update only photo
```
Comment (AntonioMrtz) [inline Backend/tests/test__playlist.py:588]: This test is checking multiple cases. Consider creating a test for every case

File: Backend/tests/test__playlist.py [hunk 6]
```diff
@@ -558,3 +559,52 @@ def test_remove_song_from_playlist_song_not_found():
...
+        name=name, descripcion=descripcion, photo=photo, headers=jwt_headers
+    )
+    assert res_create_playlist.status_code == HTTP_201_CREATED
+
+    # Update only description
+    new_description = "newdesc"
+    res_update = update_playlist_metadata(
+        name=name, descripcion=new_description, headers=jwt_headers
+    )
+    assert res_update.status_code == 204
```
Comment (AntonioMrtz) [inline Backend/tests/test__playlist.py:584]: We're using starlette HTTP codes in all the codebase. Import the corresponding code

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
